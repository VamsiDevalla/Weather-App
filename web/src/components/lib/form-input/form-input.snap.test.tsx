import React from 'react';
import { create } from 'react-test-renderer';
import FormInput from './form-input.component';

describe('Form Input', () => {
  test('should render properly', () => {
    expect(
      create(<FormInput id='test' label='snapshot test label' handleChange={jest.fn()} />).toJSON(),
    ).toMatchSnapshot();
  });
  test('should map change handle properly', () => {
    const changeHandler = jest.fn();
    const input = create(<FormInput id='test' label='snapshot test label' handleChange={changeHandler} />);
    input.root.findByType('input').props.onChange(45);
    expect(changeHandler).toHaveBeenCalledWith(45);
  });
});
