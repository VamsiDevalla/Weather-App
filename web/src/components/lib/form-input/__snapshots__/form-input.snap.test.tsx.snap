// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Input should render properly 1`] = `
<div
  className="group"
  data-testid="inputGroupContainer"
  role="none"
>
  <input
    className="form-input"
    data-testid="input"
    id="test"
    onChange={[MockFunction]}
  />
  <label
    className=" form-input-label"
    data-testid="inputLabel"
    htmlFor="test"
  >
    snapshot test label
  </label>
</div>
`;
